controller/context/finalizerskeptcontext/finalizers_kept_context.go

```diff
@@ -10,23 +10,19 @@ import (
 // collisions with keys defined in other packages.
 type key string
 
-// keptKey is the key for kept values in context.Context. Clients use
+// ctxKey is the key for kept values in context.Context. Clients use
 // finalizerskeptcontext.NewContext and finalizerskeptcontext.FromContext
 // instead of using this key directly.
-var keptKey key = "kept"
+var ctxKey key = "kept"
 
-// NewContext returns a new context.Context that carries value v.
-func NewContext(ctx context.Context, v chan struct{}) context.Context {
-	if v == nil {
-		return ctx
-	}
-
-	return context.WithValue(ctx, keptKey, v)
+// NewContext returns a new context.Context that can be used to keep finalizers.
+func NewContext(ctx context.Context) context.Context {
```

Comment:
This is a breaking change and I intend to make more the coming days and weeks, aiming for a new major version for the `operatorkit` library. We can think about what else to change. I have a couple of things in mind which we can talk about.

Reply:
Is it really breaking? Do we use it outside operatorkit?

Reply:
In a truly open source project we would need to assume that a public function may be used somewhere. I think we need to start treating the semver semantics properly.

Reply:
Yes but for us it's a version bump in `go.mod`.